requirements/py39.yml

```diff
@@ -0,0 +1,28 @@
+name: cf-units-dev
+
+channels:
+  - conda-forge
+
+dependencies:
+  - python 3.9
```

Comment:
Is there a way to create a build matrix with cirrus-ci? It seems a little brittle and hard to maintain  an env for each Python version.

Reply:
This is how ~50% of SciTools is set up now. @bjlittle is there more D.R.Y. way?

Reply:
Yeah, I'm seeing this in other projects. I wonder if going another direction with cirrus-ci is healthy for the projects in SciTools. It will be hard to get contributions from the larger SciPy community who is invested in GitHub Actions.

Reply:
> A valid point. Although I would hope that most contributions shouldn't need involvement with the CI architecture.

That is pretty much all I do since 2012!

> The caching and compute credits are appealing parts of Cirrus - faster runs.

GitHub Actions is quite fast, not sure if there are comparisons. Also, there are mechanism for caching too. I don't want to be pushy or anything. I just wonder if it is  going in another direction is worth it.

Reply:
I find this topic really interesting. I've also noticed that GHA is by far the most popular choice. And given that barriers to involvement are a big factor in the health of a project: we need a **really good** reason to go against the crowd.

From what you're saying, GHA at least has a chance of meeting the needs of SciTools projects, but we haven't tried it out yet, so at the moment we don't know if we have a really good reason to stick with Cirrus.

That said, the consensus is that SciTools and SciTools-incubator projects should have a consistent setup, so this conversation is bigger than this PR. I've therefore raised an issue against Iris - as the project with the largest test suite - to pursue this further: https://github.com/SciTools/iris/issues/4191. In the meantime this question shouldn't prevent this PR being merged IMHO.

Reply:
Looks like Cartopy is using GHA, so not completely alien to SciTools.